### PR TITLE
Working catalog type filter

### LINKED ISSUE #1256: Working type filters on catalog page

### PR DESCRIPTION
**Pull request by hooloovooo**
## Hey, I just made a Pull Request!

Filtering by type tabs on catalog page

2020-06-16-130522_961x354_scrot

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by hooloovooo** [comment]
Filtering works in this PR, but the count number on the owned/starred/all filters aren't affected by this. Adding this draft for early feedback.

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx [hunk 1]
```diff
@@ -71,16 +97,22 @@ export const CatalogPage: FC<{}> = () => {
...
     [],
   );
 
+  const filteredEntities = useMemo(() => {
+    const typeFilter = entityFilters[EntityFilterType.TYPE];
+    const leftMenuFilter = entityFilters[selectedFilter.id];
+    return value
+      ?.filter(e => leftMenuFilter(e, { isStarred: isStarredEntity(e) }))
+      .filter(e => typeFilter(e, { type: selectedTab }));
+  }, [selectedFilter.id, selectedTab, isStarredEntity, value?.filter]);
```
Comment (shmidt-i) [inline plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:114]: `value.filter` feels wrong here conceptually, maybe just `value`? Like in "if value changes, recalculate this"

Comment (hooloovooo) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:114]: > `value.filter` feels wrong here conceptually, maybe just `value`? Like in "if value changes, recalculate this"

Yea i agree. The linter hated that though. This exhaustive-deps rule is weird and i don't understand it fully. Leaving it out suggested me adding what's there now, and adding just value as you suggested gave this error `React Hook useMemo has an unnecessary dependency: 'value'. Either exclude it or remove the dependency array`

**Comment by benjdlambert** [comment]
So, I'm not sure if I understand #983 correctly, but I think the plan is to move the `type` field to the `metadata` block in the  entity definition. @freben  is that correct or have I misunderstood?

If that's true, I wonder if it's worth doing that part of this work, so we can use the tag filter stuff that we built in the `catalog-backend` API to filter on the metadata fields rather than grabbing all the entities for every filter, tabs and filters on the side of the table.

I know that it might be an early optimisation, and graphql might solve it, but I think were we can make things like this better and push the logic further down the stack we should.

**Comment by freben** [comment]
No, the `type` field is not considered for moving.

But if you use the backend for filtering, you don't have to care. All fields everywhere are reachable without prefixes. So you can use `?metadata.annotations.blah=`, `?annotations.blah=` and `?blah=` and you would get the exact same results.

So just go for `?type=` either way and you'll be fine.

**Comment by benjdlambert** [comment]
Oh ok apologies!, I just thought only metadata fields were indexed. In that case why don’t we just use the filter endpoint and request the data we need with the filters to set the type over requesting all and filtering client side

**Comment by freben** [comment]
I think that will work for now. With the drawback of requiring round trips to the server of course.

**Comment by hooloovooo** [comment]
> No, the `type` field is not considered for moving.
> 
> But if you use the backend for filtering, you don't have to care. All fields everywhere are reachable without prefixes. So you can use `?metadata.annotations.blah=`, `?annotations.blah=` and `?blah=` and you would get the exact same results, for example.
> 
> So just go for `?type=` either way and you'll be fine.

Do you think using backend for filtering type is a better solution?

**Comment by freben** [comment]
Well do what's necessary to get this in first. If going to the backend is an easiler solution then do that. Otherwise keep the client side filtering.

**Comment by hooloovooo** [comment]
> Well do what's necessary to get this in first. If going to the backend is an easiler solution then do that. Otherwise keep the client side filtering.

They're both very easy. It's probably cleaner to keep all filtering in one place though.

**Review by stefanalund — CHANGES_REQUESTED** [review]
@hooloovooo there are inconsistencies in the labels:

Screenshot 2020-06-16 at 11 20 03

Also the main title still says "Services" even though I switch to Websites or other tabs.

**Review by stefanalund — APPROVED** [review]
Let's get this in and create issues for the remaining work!

**Pull request merged**